### DISCUSSION TITLE
How to refer to a column after join operation

### DISCUSSION THREAD

**Discussion by Androidown** [PRQL/prql#5052] [Q&A]
To say, I have a relation which is created from a join operation.
```elm
from a=invoices
join b=customers (a.invoice_id == b.customer_id)
```
I want to select column `field` of that relation, but somehow I don't know which relation `field` belongs to, what I do know is column `field` does exist in that joined relation.
I was expecting the following prql to compile.
```elm
from a=invoices
join b=customers (a.invoice_id == b.customer_id)
select {field}
```
Expected SQL:
```sql
with table_0 as (
SELECT
  a.*,
  b.*
FROM
  invoices AS a
  JOIN customers AS b ON a.invoice_id = b.customer_id
)
SELECT
   field
FROM
  table_0 
```

But it fails with error:
```
Error: 
   ╭─[:3:9]
   │
 3 │ select {field}
   │         ──┬──  
   │           ╰──── Ambiguous name
   │ 
   │ Help: could be any of: a.field, b.field
───╯
```

**Comment by richb-hanover** [comment]
I'm a SQL newcomer, but have been using PRQL for a while... As I think about this problem, the `field` column must have come from either table `a` or `b`. So I think that all you need to do is use `select { a.field }` or `select { b.field }` to indicate which table originally contained the `field` column.

Let us know if that answers the question (or not!) Thanks

**Reply by richb-hanover** [reply]
@Androidown I realize that my earlier response doesn't really address your question. (Sorry.)

But I am not really sure that I understand your use case. Could you tell more about your application that causes it not to be able to tell whether `field` comes from table `a` or `b`? Thanks
